### PR TITLE
add ut for shp:fill and new DV constructors

### PR DESCRIPTION
**Pull request by haichangsi**

### CODE REVIEW

**Review by rscohn2 — APPROVED** [review]

File: test/gtest/shp/containers.cpp [hunk 1]
```diff
@@ -26,6 +26,22 @@ TYPED_TEST(DistributedVectorTest,
   EXPECT_EQ(second[0], lib::ranges::segments(second)[0][0]);
 }
 
+TYPED_TEST(DistributedVectorTest, fill_constructor) {
+  EXPECT_TRUE(equal(typename TestFixture::DistVec(10, 1),
+                    typename TestFixture::LocalVec(10, 1)));
+}
+
+// TODO: add more tests: fill_constructor_one_item, fill_constructor_no_items
```
Comment (lslusarczyk) [inline test/gtest/shp/containers.cpp:34]: @haichangsi , as an exercise of writing concise UTs please implement this TODO and TODO below

**Review by lslusarczyk — CHANGES_REQUESTED** [review]

File: test/gtest/shp/containers.cpp [hunk 2]
```diff
@@ -31,16 +31,32 @@ TYPED_TEST(DistributedVectorTest, fill_constructor) {
                     typename TestFixture::LocalVec(10, 1)));
 }
 
-// TODO: add more tests: fill_constructor_one_item, fill_constructor_no_items
+TYPED_TEST(DistributedVectorTest, fill_constructor_one_item) {
+  EXPECT_TRUE(equal(typename TestFixture::DistVec(1),
```
Comment (lslusarczyk) [inline test/gtest/shp/containers.cpp:35]: this ctor initializes with random value, both vectors may be different
use ctor with explicit value, e.g DistVec(1,77), LocalVec(1, 77)

**Review by lslusarczyk — CHANGES_REQUESTED** [review]

**Review by lslusarczyk — APPROVED** [review]

**Comment by rscohn2** [comment]
@lslusarczyk Can these be made into shared tests (mhp/shp)? mhp does not support initializer list in the constructor. That would be a good thing to add.

**Comment by lslusarczyk** [comment]
sure, created https://github.com/oneapi-src/distributed-ranges/issues/104 for this

**Pull request merged**